npm.d.ts

```diff
@@ -0,0 +1,439 @@
+
+	/**
+	Run with the `npm test` command.
+	*/
+	test?: string;
+
+	/**
+	Run with the `npm test` command, after `test`.
+	*/
+	posttest?: string;
+
+	/**
+	Run with the `npm stop` command, before `stop`.
+	*/
+	prestop?: string;
+
+	/**
+	Run with the `npm stop` command.
+	*/
+	stop?: string;
+
+	/**
+	Run with the `npm stop` command, after `stop`.
+	*/
+	poststop?: string;
+
+	/**
+	Run with the `npm start` command, before `start`.
+	*/
+	prestart?: string;
+
+	/**
+	Run with the `npm start` command.
+	*/
+	start?: string;
+
+	/**
+	Run with the `npm start` command, after `start`.
+	*/
+	poststart?: string;
+
+	/**
+	Run with the `npm restart` command, before `restart`. Note: `npm restart` will run the `stop` and `start` scripts if no `restart` script is provided.
+	*/
+	prerestart?: string;
+
+	/**
+	Run with the `npm restart` command. Note: `npm restart` will run the `stop` and `start` scripts if no `restart` script is provided.
+	*/
+	restart?: string;
+
+	/**
+	Run with the `npm restart` command, after `restart`. Note: `npm restart` will run the `stop` and `start` scripts if no `restart` script is provided.
+	*/
+	postrestart?: string;
+
+	[scriptName: string]: string | undefined;
+}
+
+/**
+Names of package dependencies that are bundled when the package is publihed.
+*/
+export type BundledDependency = string[];
+
+/**
+Dependencies of the package. The version range is a string which has one or more space-separated descriptors. Dependencies can also be identified with a tarball or Git URL.
+*/
+export interface Dependency {
+	[packageName: string]: string;
+}
+
+export type PackageJson = {
+	/**
+	The name of the package.
+	*/
+	name?: string;
+
+	/**
+	Package version, parseable by [`node-semver`](https://github.com/npm/node-semver).
+	*/
+	version?: string;
+
+	/**
+	Package description, listed in `npm search`.
+	*/
+	description?: string;
+
+	/**
+	Keywords associated with package, listed in `npm search`.
+	*/
+	keywords?: string[];
+
+	/**
+	The URL to the package's homepage.
+	*/
+	homepage?: LiteralUnion<'.', string>;
+
+	/**
+	The URL to the package's issue tracker and/or the email address to which issues should be reported.
+	*/
+	bugs?: BugsLocation;
+
+	/**
+	The license for the package.
+	*/
+	license?: string;
+
+	/**
+	The licenses for the package.
+	*/
+	licenses?: {
+		type?: string;
+		url?: string;
+	}[];
+
+	author?: Person;
+
+	/**
+	A list of people who contributed to the package.
+	*/
+	contributors?: Person[];
+
+	/**
+	A list of people who maintain the package.
+	*/
+	maintainers?: Person[];
+
+	/**
+	The files included in the package.
+	*/
+	files?: string[];
+
+	/**
+	The module ID that is the primary entry point to the program.
+	*/
+	main?: string;
+
+	bin?:
+	| string
+	| {
+		[binary: string]: string;
+	};
+
+	/**
+	Location of the bundled TypeScript declaration file.
+	*/
+	types?: string;
+
+	/**
+	Location of the bundled TypeScript declaration file. Synonymous with `types`.
+	*/
+	typings?: string;
```

Comment:
These are not official fields. What do you think about putting them in https://github.com/sindresorhus/type-fest/pull/5/files#diff-6c293fd7802982e4829c46838e706800R438 or a separate type for clarity?

Reply:
Can you also add the `browser` field there? https://github.com/defunctzombie/package-browser-field-spec